sphericart/include/sphericart.h

```diff
@@ -13,8 +13,10 @@
 
 #include "sphericart.hpp"
 
-using sphericart_calculator_t = sphericart::SphericalHarmonics<double>;
-using sphericart_calculator_f_t = sphericart::SphericalHarmonics<float>;
+using sphericart_spherical_harmonics_calculator_t = sphericart::SphericalHarmonics<double>;
+using sphericart_spherical_harmonics_calculator_f_t = sphericart::SphericalHarmonics<float>;
+using sphericart_solid_harmonics_calculator_t = sphericart::SolidHarmonics<double>;
+using sphericart_solid_harmonics_calculator_f_t = sphericart::SolidHarmonics<float>;
```

Comment:
Do we really need separate types here? We could reduce the number of functions by either having a single calculator type, which is initialized to be either `SphericalHarmonics` or `SolidHarmonics`, or having two calculator types by a single function that can take both and call the right function on them.

If this is not an issue, we could also keep the two separate, but I feel like we have a lot of functions in the C API for no reason.

Reply:
I agree that we have a lot of functions (and we would have even more if we wanted to have a CUDA C API for e.g. CuPy). However, I feel like this is in line with the other APIs and the original idea behind this PR was to make the APIs as uniform as possible

Reply:
Thanks. And yes, let's discuss in person